source/jormungandr/jormungandr/utils.py

```diff
@@ -458,6 +458,8 @@ def decode_polyline(encoded, precision=6):
 
 class SectionSorter(object):
     def __call__(self, a, b):
+        if a.begin_date_time == b.begin_date_time and a.end_date_time == b.end_date_time:
+            return -1 if a.destination.uri == b.origin.uri else 1
         if a.begin_date_time != b.begin_date_time:
             return -1 if a.begin_date_time < b.begin_date_time else 1
         else:
```

Comment:
Smart to try to join sections :bulb: 
But how about 3 sections having the same values, or 2 "distant" sections? Maybe raising would be better :shrug: 

Anyway, more efficient (and clearer IMO):
```py
        if a.begin_date_time != b.begin_date_time:
            return -1 if a.begin_date_time < b.begin_date_time else 1
        elif a.end_date_time == b.end_date_time:
           return -1 if a.end_date_time < b.end_date_time else 1
        elif a.destination.uri == b.origin.uri:
           return -1
        elif a.origin.uri == b.destination.uri:
           return 1
        else:  # still not decided, but default to something stable. Or is there a return 0 possible ?
           return -1 if a.origin.uri < b.origin.uri else 1  # or should we explicitly raise?
```

Reply:
I like your proposition better, it's clearer. However, about you last comment, if I'm not wrong, I wouldn't recommend to return `0` (even if it's possible) because we want a stable sorting and depending on the sorting algorithm, order might or might not be preserved. By the way, this class is called `SectionSorter` but I don't think the naming is correct, it doesn't sort, it only compares. How about renaming it `SectionComparator`?

Reply:
Now I am wondering why we need to sort sections :thinking:

Reply:
@pbench In the protobuf you cannot add a section at the beginning

Reply:
@woshilapin returning a raw value in the else case (no matter the value) is not stable, as it will depend on which is left or right member. Sorting on uri is stable but not relevant... I guess raising has my preference, but it's not strong.
:heavy_check_mark: Anyway, it's quite clear and way better now, so it's fine by me.

Naming proposition by @woshilapin is better but not mandatory IMO.
@pbench avoid that sorting by a stitching mechanism of section-groups would be more correct, but this is more work, so I'm OK to keep this.